Centaurus.Domain/Extensions/ExtensionsManager.cs

```diff
@@ -59,40 +59,40 @@ public async Task RegisterAllExtensions()
         public event EventHandler<BaseWebSocketConnection> OnConnectionValidated;
         public void ConnectionValidated(BaseWebSocketConnection args)
         {
-            OnConnectionValidated?.Invoke(this, args);
+            OnConnectionValidated?.Invoke(null, args);
```

Comment:
When I was saying "get rid of `this` in the arguments", I meant "use some other kind of delegate" in the first place. 
For instance, you can define a delegate like
```c#
public delegate void ExtensionEvent<T>(T args);
public delegate void ExtensionEvent<T1, T2>(T1 arg1, T2 arg2);
public delegate void ExtensionEvent<T1, T2, T3>(T1 arg1, T2 arg2, T3 arg3);
```
and then use it to declare any event as you like:

```c#
//one arg
public event ExtensionEvent<BaseWebSocketConnection> OnConnectionValidated;
//two args
public event ExtensionEvent<BaseWebSocketConnection, MessageEnvelope> OnBeforeSendMessage;
```